src/amp-story-player/amp-story-player-impl.js

```diff
@@ -730,20 +771,79 @@ export class AmpStoryPlayer {
 
   /**
    * Indicates the player changed story.
+   * @param {!Object} data
    * @private
    */
-  signalNavigation_() {
+  signalNavigation_(data) {
+    const event = createCustomEvent(this.win_, 'navigation', data);
+    this.element_.dispatchEvent(event);
+  }
+
+  /**
+   * Triggers when swithing from one story to another.
+   * @private
+   */
+  onNavigation_() {
     const index = this.currentIdx_;
     const remaining = this.stories_.length - this.currentIdx_ - 1;
-    const event = createCustomEvent(
-      this.win_,
-      'navigation',
-      dict({
-        'index': index,
-        'remaining': remaining,
-      })
-    );
-    this.element_.dispatchEvent(event);
+    const navigation = {
+      'index': index,
+      'remaining': remaining,
+    };
+
+    this.signalNavigation_(navigation);
+    this.maybeFetchMoreStories_(navigation);
+  }
+
+  /**
+   * Fetches more stories if appropiate.
+   * @param {!Object} data
+   */
+  maybeFetchMoreStories_(data) {
```

Comment:
This `data` parameter will no longer make sense if you ever have to call `maybeFetchMoreStories_` from a non navigation context (eg: you wouldn't have the `navigation` object). If the method only needs a `remaining` parameter we should make it more clear.
That'll also allow you to write more descriptive documentation: `{number} remaining` makes your method easier to use than `{!Object} data` that could be anything.